registrations/models.py

```diff
@@ -581,6 +589,28 @@ def move_first_waitlisted_to_attending(self, first_on_list=None):
                 SignUpNotificationType.TRANSFERRED_AS_PARTICIPANT
             )
 
+    def cancel_signups(self, is_event_cancellation=False):
+        for signup_group in self.signup_groups.all():
+            signup_group.delete(is_event_cancellation=is_event_cancellation)
+
+        for signup in self.signups.filter(signup_group_id__isnull=True):
+            signup._individually_deleted = True
+            signup.delete(is_event_cancellation=is_event_cancellation)
```

Comment:
I am wondering, has the impact of these loops failing halfway been evaluated sufficiently? 

The `save` which triggers this process has `@transaction.atomic` ( https://github.com/City-of-Helsinki/linkedevents/pull/901/files#diff-e22db1fd6b46c1799d0bb96df4d30720dba88c4ab310face6db5d9f9379a473cR1117 ). 

Therefore, if one of these deletes fail, don't we end up rolling back the whole save transaction?

And suppose that some deletes/refunds were completed in the web store before the failure, then our database is left in an inconsistent state?

Reply:
True, if this fails after a successful refund or cancellation request, then there will a inconsistent state between Talpa and LE and successfully refunded or cancelled signups or groups will still be there in the LE DB. While this was considered, the point of that happening after multiple successful requests was not perhaps evaluated sufficiently enough.

So perhaps we are better off running these loops outside the transaction and allowing individual signups and groups to be deleted after successful requests to Talpa, stopping to an exception if that happens.

Reply:
I'll let you decide the best course. It's of course very hard to be 100% foolproof, even if this is done outside of transaction, say a request fails with 502 gateway timeout, we have no idea if the refund ends up being processed or not.

Reply:
Do you still want to address this further in the scope of this PR? Otherwise this looks fine to me and I can stamp it.

Reply:
I will address this in this ticket soon :-) Just been a bit too busy with another larger development

Reply:
@voneiden `event.save()` was changed so that in case of payments, the signup cancellation happens outside of the event's atomic transaction. Otherwise, if there are no payments, the code will function as it did before.